tools/data_loader.py

```diff
@@ -0,0 +1,378 @@
+                    group_ids = []
+                    group_dict = {}
+                    filter_dict = {}
+                    for g in src.get('groups', []):
+                        data = assert_post(
+                            "groups",
+                            {
+                                "name": g["name"],
+                                "group_admins": g["group_admins"]
+                            },
+                            tokens[g["token"]]
+                        )
+                        group_ids.append(data["data"]["id"])
+                        group_dict[g["name"]] = group_ids[-1]
+
+                        for f in g.get("filters", []):
+                            data = assert_post(
+                                "filters",
+                                {
+                                    "group_id": group_ids[-1],
+                                    "query_string": f["query_string"]
+                                },
+                                tokens[g["token"]]
+                            )
+                            filter_dict[f["name"]] = data["data"]["id"]
+
+                        for m in g.get("members", []):
+                            data = assert_post(
+                                f"groups/{group_ids[-1]}/users/{m['username']}",
+                                {"admin": m.get('admin', False)},
+                                tokens[g["token"]]
+                            )
+            if src.get("taxonomies") is not None:
+                with status("Creating Taxonomies"):
+
+                    for tax in src.get('taxonomies', []):
+                        name = tax["name"]
+                        provenance = tax.get("provenance")
+
+                        if tax["tdtax"]:
+                            hierarchy = tdtax.taxonomy
+                            version = tdtax.__version__
+                        else:
+                            hierarchy = tax["hierarchy"]
+                            tdtax.validate(hierarchy, tdtax.schema)
+                            version = tax["version"]
+
+                        group_ids = [group_dict[g] for g in tax["groups"]]
+                        payload = {
+                            "name": name,
+                            "hierarchy": hierarchy,
+                            "group_ids": group_ids,
+                            "provenance": provenance,
+                            "version": version,
+                        }
+                        data = assert_post(
+                            "taxonomy",
+                            payload,
+                            tokens[tax["token"]]
+                        )
+            if src.get("telescopes") is not None:
+                with status("Creating Telescopes"):
+                    telescope_dict = {}
+                    tel_src = src.get("telescopes")
+                    if isinstance(tel_src, dict) and \
```

Comment:
Should we be catching the case where this is `False` and `group_names`, `token`, etc below are `None`? Currently it'll raise `NameError` but perhaps we should have a more descriptive error message, e.g. "This value must be specified either here or here" (where we describe the possible configurations)

Reply:
Same for other data types below where relevant (instruments, etc)

Reply:
It's fine for this to throw errors -- this is a one-time run script (at the beginning of a SkyPortal instance) and not production code. There's another issue for doing YAML schema checking which will help with malformed YAMLs.